cubedash/_stac.py

```diff
@@ -0,0 +1,235 @@
+import logging
+
+from collections import OrderedDict
+from flask import Blueprint, request, abort, url_for
+
+from datacube.api.query import Query
+from datacube.model import Range
+from datacube.utils.geometry import CRS, Geometry
+
+import datetime
+from dateutil.parser import parse
+from pandas import Timestamp, DateOffset
+import yaml
+import json
+import itertools
+
+from . import _utils as utils
+from . import _model
+
+
+_LOG = logging.getLogger(__name__)
+bp = Blueprint('stac', __name__, url_prefix='/stac')
+
+CFG = yaml.load(
+    """
+    homepage: "http://www.ga.gov.au/"
+
+    license:
+      name: "Creative Commons Attribution 4.0 International"
+      link: "https://spdx.org/licenses/CC-BY-4.0.html#licenseText"
+      short_name: "CC-BY-4.0"
+      copyright: "DEA, Geoscience Australia"
+
+    contact:
+      name: "Geoscience Australia"
+      organization: "Commonwealth of Australia"
+      email: "sales@ga.gov.au"
+      phone: "+61 2 6249 9966"
+      url: "http://www.ga.gov.au"
+    """
+)
+
+MAX_DATASETS = 100
+DATASETS_PER_REQUEST = 20
+
+
+@bp.route('/')
+def root():
+    return utils.as_json({})
+
+
+@bp.route('/search', methods=['GET', 'POST'])
+def stac_search():
+    if request.method == 'GET':
+        bbox = request.args.get('bbox')
+        time_ = request.args.get('time')
+        product = request.args.get('product')
+        limit = request.args.get('limit')
+        from_dts = request.args.get('from')
+    else:
+        req_data = request.get_json()
+        bbox = req_data.get('bbox')
+        time_ = req_data.get('time')
+        product = req_data.get('product')
+        limit = req_data.get('limit')
+        from_dts = req_data.get('from')
+
+    # bbox and time are compulsory
+    if not bbox:
+        abort(404, "bbox must be specified")
```

Comment:
Use status code 400 for bad arguments, not 404. (there's a few more in this file to change too)
```suggestion
        abort(400, "bbox must be specified")
```

See the list of different codes here: https://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml
(or more friendly: https://httpstatuses.com/)

(Note that `bbox` and `time` are optional in the stac spec, but we can implement that later.)